x/subscription/keeper/subscription.go

```diff
@@ -230,80 +236,86 @@ func (k Keeper) CreateSubscription(
 	return err
 }
 
+func (k Keeper) createNewSubscription(ctx sdk.Context, plan *planstypes.Plan, creator, consumer string,
+	block uint64, autoRenewalFlag bool,
+) (types.Subscription, error) {
+	sub := types.Subscription{
+		Creator:       creator,
+		Consumer:      consumer,
+		Block:         block,
+		PlanIndex:     plan.Index,
+		PlanBlock:     plan.Block,
+		DurationTotal: 0,
+		AutoRenewal:   autoRenewalFlag,
+	}
+
+	sub.MonthCuTotal = plan.PlanPolicy.GetTotalCuLimit()
+	sub.MonthCuLeft = plan.PlanPolicy.GetTotalCuLimit()
+	sub.Cluster = types.GetClusterKey(sub)
+
+	// new subscription needs a default project
+	err := k.projectsKeeper.CreateAdminProject(ctx, consumer, *plan)
+	if err != nil {
+		return types.Subscription{}, utils.LavaFormatWarning("failed to create default project", err)
+	}
+
+	return sub, nil
+}
+
+func (k Keeper) upgradeSubscriptionPlan(ctx sdk.Context, duration uint64, sub *types.Subscription, toPlanIndex string, toPlanBlock uint64) error {
+	date := ctx.BlockTime()
+	block := uint64(ctx.BlockHeight())
+
+	k.addCuTrackerTimerForSubscription(ctx, block, sub)
```

Comment:
add a comment, this is for the last subscription

Reply:
Fixed in 0d7711ddca5b3cbfcd393fdcb7cbd60e8b2573b5